rust/lance/src/dataset/write/merge_insert.rs

```diff
@@ -1354,6 +1401,7 @@ impl MergeInsertJob {
                 updated_fragments,
                 new_fragments,
                 fields_modified,
+                mem_wal_to_flush: self.params.mem_wal_to_flush,
```

Comment:
At some point we will need to read the mem wal and write the data correct?  Is that a TODO?